pkg/policies/controlplane/runtime/circuit.go

```diff
@@ -208,25 +221,26 @@ func (circuit *Circuit) Execute(tickInfo TickInfo) error {
 			PolicyHash:    circuit.GetPolicyHash(),
 			SignalReading: make([]*policymonitoringv1.SignalReading, 0),
 		}
+		var invalidCount int
 		// log all circuitSignalReadings
 		for signal, reading := range circuitSignalReadings {
+			circuitMetricsLabels := prometheus.Labels{
+				metrics.SignalNameLabel:   signal.SignalName,
+				metrics.SubCircuitIDLabel: signal.SubCircuitID,
+				metrics.PolicyNameLabel:   circuit.Policy.GetPolicyName(),
+			}
+			if !reading.Valid() {
+				circuitMetrics.SignalSummaryVec.Delete(circuitMetricsLabels)
+				invalidCount++
+				continue
```

Comment:
Can't continue here. Status needs to be updated in the loop for the signal heartbeats to work for both valid and invalid signals